src/use_file.rs

```diff
@@ -93,6 +115,25 @@ fn wait_until_rng_ready() -> Result<(), Error> {
     }
 }
 
+// Succeeds once /dev/urandom is safe to read from
+#[cfg(all(any(target_os = "android", target_os = "linux"), feature = "rustix"))]
+fn wait_until_rng_ready() -> Result<(), Error> {
+    use rustix::event;
+
+    // Open the file.
+    let fd = crate::util_rustix::open_readonly("/dev/random")?;
+
+    // Poll it until it is ready.
+    let mut pfd = [event::PollFd::new(&fd, event::PollFlags::IN)];
+    loop {
+        match event::poll(&mut pfd, -1) {
+            Ok(_) => return Ok(()),
+            Err(rustix::io::Errno::INTR) => continue,
+            Err(err) => return Err(crate::util_rustix::cvt(err)),
+        }
+    }
+}
+
 struct Mutex(UnsafeCell<libc::pthread_mutex_t>);
```

Comment:
The most important improvement rustix could make to this code would be replacing the pthreads_mutex_t here with whatever is used by libstd on the target (e.g. futex on Linux/Android).

Reply:
Good point. I've replaced `Mutex` with one based on futexes, only on Rustix.